### PR TITLE
Attributes view list

### PR DESCRIPTION
**Pull request by SanaBob**
Create API to fetch all attributes for the profiles page

### CODE REVIEW

**Comment by SanaBob** [comment]
In the spec, you mentioned we needed `name` for attribute return data which I'm not sure what it was or needed. Also, for attribute options, do we need to remove fields such as timestamps or we can just leave fields as all?

File: app/views/attribute.py [hunk 1]
```diff
@@ -7,3 +8,16 @@
 class AttributeViewSet(viewsets.ModelViewSet):
     queryset = Attribute.objects.all()
     serializer_class = AttributeSerializer
+    filter_backends = []
+
+    def list(self, request):
+        course_id = request.GET.get("course_id", None)
+        query_set = self.filter_queryset(self.get_queryset())
```
Comment (Opeyem1a) [inline app/views/attribute.py:15]: I feel like there is a correct way to do this in Django REST
Look at this page - https://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend

File: app/views/attribute.py [hunk 2]
```diff
@@ -7,3 +8,16 @@
...
+
+        if course_id:
+            query_set = query_set.filter(course_id=course_id)
+        else:
+            return Response("course_id is required", status=status.HTTP_400_BAD_REQUEST)
```
Comment (Opeyem1a) [inline app/views/attribute.py:20]: Not really the correct error here. Auth hasn't been done so not within scope, but ideally here the flow would be:
- Admin users can get all attributes
- Normal users can only request from courses they are the instructor of or a student in

File: app/serializers/attribute.py [hunk 3]
```diff
@@ -14,7 +14,13 @@ class AttributeSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Attribute
-        fields = "__all__"
+        fields = [
```
Comment (Opeyem1a) [inline app/serializers/attribute.py:17]: I think it's fine to leave this as all for now

**Review by Opeyem1a — CHANGES_REQUESTED** [review]
It's fine to send more data than you need in the early days of this backend, I don't think it's a big thing to worry about. (Regarding the timestamps and id on AttributeOption)

Note though that you will definitely need the AtrributeOption id to properly update them

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**